Include/Rover/Constant.hpp

```diff
@@ -0,0 +1,60 @@
+#ifndef ROVER_CONSTANT_HPP
+#define ROVER_CONSTANT_HPP
+#include <type_traits>
+#include "Generator.hpp"
+#include "Noncopyable.hpp"
+
+namespace Rover {
+
+  /** An argument generator that evaluates to a constant value. */
+  template<typename T>
+  class Constant : Noncopyable {
+    static_assert(!implements_concept_v<T, Generator>,
```

Comment:
The issue here, and with the entire generator "concept", is that creating a Constant using a Generator yields slightly unexpected results. For example,
// Good
auto r1 = make_reference<Range>(make_reference<Constant>(1), make_reference<Constant>(10));
// or auto r1 = Reference(Range(Constant(1), Constant(10)));
auto r2 = make_reference<Range>(make_reference<Constant>(r1), make_reference<Constant>(20));
auto gen = make_reference<Composite>(r1, r2);
// generate(gen) yields x1, x2 such as x2 >= x1

// Bad
auto r1 = Range(Constant(1), Constant(10));
auto r2 = Range(Constant(r1), Constant(20));
auto gen = Composite(r1, r2); // currently disabled using static_assert
// generate(get) will yield independent x1 and x2 as r1 generator (not a reference) was copied to the r2 begin 

Range handles the same issue by casting each generator to a reference (which in turn requires it be an rvalue to prevent free generator re-use), but I cannot do the same for Constant as it must be able to handle arbitrary types - otherwise we won't be able to create any Generator to begin with.

I only see 2 possible solutions:
1. Just remove the assertion and make not creating Constant(Generator) a user's responsibility - may interfere with some generic programming where Generators and arbitrary types are treated the same.
2. Disable construction of non-reference Generators and only permit the perfect forwarding syntax. This will also make handling free Generators in other Generators unnecessary and hence much easier to define a new one, especially in Python.

Reply:
P.S. It does not display template braces in the message, click on Edit to view the full valid code.